### DISCUSSION TITLE
To what extent does Minestom validate serverbound packets?

### DISCUSSION THREAD

**Discussion by rj00a** [Minestom/Minestom#1527] [General]
Hello. To what extent does Minestom validate serverbound packets? Would a hacker be able to do any of these things without the programmer explicitly checking for it?

1. Fly/teleport arbitrary distances
2. Teleport through walls
3. Interact with entities further away than they should
4. Sneak and sprint at the same time
5. Move while an inventory is open
6. Break blocks faster than expected

I'm wondering what Minestom's stance is on such issues. I ask because I'm debating where the line should be drawn for [my own server implementation](https://github.com/valence-rs/valence).

**Comment by MrGeoTech** [comment]
I'm not entirely sure as I haven't looked through all the code but I have tried hacked clients on Minestom and almost all the hacks work so I assume there is no check/minimal check. I know that you can fly and use speed. You can do your own testing too.
